src/vdaf/prio3.rs

```diff
@@ -196,12 +208,73 @@ fn check_num_aggregators(num_aggregators: u8) -> Result<(), VdafError> {
     Ok(())
 }
 
-/// The base type for prio3.
+/// The base type for Prio3.
+///
+/// An instance of Prio3 is determined by:
+///
+/// - a [`Type`](crate::flp::Type) that defines the set of valid input measurements;
+/// - a [`Prg`](crate::vdaf::prg::Prg) for deriving vectors of field elements from seeds; and
+/// - a method for decoding the aggregate result from a vector of field elements.
+///
+/// New instances can be defined by aliasing the base type. For example, [`Prio3Aes128Count`] is an
+/// alias for `Prio3<Count<Field64>, Prio3Result<u64>, PrgAes128, 16>`.
+///
+/// ```
+/// use prio::vdaf::{
+///     Client, Aggregator, Collector,
+///     prio3::{Prio3Result, Prio3Aes128Count}
+/// };
+/// use rand::prelude::*;
+///
+/// let num_shares = 2;
+/// let vdaf = Prio3Aes128Count::new(num_shares).unwrap();
+///
+/// let mut out_shares = vec![vec![]; num_shares.into()];
+/// let mut rng = thread_rng();
+/// let verify_key = rng.gen();
+/// for measurement in [0, 1, 1, 1, 0] {
+///     // Shard
+///     let input_shares = vdaf.shard(&measurement).unwrap();
+///     let mut nonce = [0; 16];
+///     rng.fill(&mut nonce);
+///
+///     // Prepare
+///     let mut prep_states = vec![];
+///     let mut prep_shares = vec![];
+///     for (agg_id, input_share) in input_shares.iter().enumerate() {
+///         let (state, share) = vdaf.prepare_init(
+///             &verify_key,
+///             agg_id,
+///             &(),
+///             &nonce,
+///             input_share
+///         ).unwrap();
+///         prep_states.push(state);
+///         prep_shares.push(share);
+///     }
+///     let prep_msg = vdaf.prepare_preprocess(prep_shares).unwrap();
+///
+///     for (agg_id, state) in prep_states.into_iter().enumerate() {
+///         out_shares[agg_id].push(vdaf.prepare_step(state, prep_msg.clone())
+///             .unwrap().unwrap_finish());
+///     }
+/// }
+///
+/// // Aggregate
+/// let agg_shares = out_shares.into_iter()
+///     .map(|o| vdaf.aggregate(&(), o).unwrap());
+///
+/// // Unshard
+/// let agg_res = vdaf.unshard(&(), agg_shares).unwrap();
+/// assert_eq!(agg_res, Prio3Result(3));
+/// ```
+///
+/// [draft-irtf-cfrg-vdaf-01]: https://datatracker.ietf.org/doc/draft-irtf-cfrg-vdaf/01/
 #[derive(Clone, Debug)]
 pub struct Prio3<T, A, P, const L: usize>
 where
     T: Type,
-    A: Clone + Debug,
+    A: TryFrom<AggregateShare<T::Field>, Error = VdafError>,
```

Comment:
Why did we lose the `Clone + Debug` trait bound here?

Reply:
No reason, added back.